### DISCUSSION TITLE
Datatable pagination links do not work - when model caching is enabled

### DISCUSSION THREAD

**Discussion by mscott83** [livewire/livewire#3191] [7. Bugs (archived)]
### Description
This is a continuation of #1775. I figured out how to make it reproducible.

I've implemented model caching on the model for which I'm trying to create a datatable. I'm using Genealabs package here: https://github.com/GeneaLabs/laravel-model-caching

When I enable model caching using the above package, that breaks the pagination. Disable the caching, and it works fine.

### Exact steps to reproduce

See #1775 for the component code, but on the model if you enable caching like so:
```
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Laravel\Scout\Searchable;
use GeneaLabs\LaravelModelCaching\Traits\Cachable;

class Airbase extends Model
{
    use HasFactory, Searchable, Cachable;
}
```
then the pagination links stop working. Switch back to the default, non-cached model like so:
```
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Laravel\Scout\Searchable;

class Airbase extends Model
{
    use HasFactory, Searchable;
}
```
and it's fine again.

I'd really rather not disable caching on every single model that I want to create datatables for. Is there any way to get Livewire to work with this? Or is it a requirement to use the standard Laravel model type? 

**Context**
- Livewire version: 2.3.0
- Laravel version: 8.4.0
- Alpine version: [e.g. 2.3.0]
- Browser: Chrome

**Comment by uteq** [comment]
I looked into this a bit and it appears someone also reported this on the model caching repo

https://github.com/GeneaLabs/laravel-model-caching/issues/377

Further I tried to unravel from the model cache repo if there is anything that could interfere with pagination. My best guess is that because Livewire is using a different request on any change in the component somehow the cache gets screwed.

Could you elaborate on what actually breaks when using the pagination?

**Comment by mscott83** [comment]
Good to see I'm not the only one, I thought I was going mad for a while.

The problem is that the pagination buttons don't do anything - the table loads with the page 1 results, but if you click any page 2/3/4/5 etc the results do not change, only the first page's results are ever shown.

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)

**Comment by jimmydinamika** [comment]
REOPEN
